### DISCUSSION TITLE
__webpack_hash__ is different than `[hash]` placeholder

### DISCUSSION THREAD

**Discussion by mborysenko** [webpack/webpack#16714] [Q&A]
I'm trying to inject `[hash]` I use for the file generation into my runtime. I use `__webpack_hash__` variable but it contains totally different value than the filenames in artefacts. Is it correct usage? What may affect that hash?
Webpack rule:
```ts
{
       ....
      {
        test: /\.json$/,
        include: path.resolve('src/client/locales'),
        type: 'asset/resource',
        generator: {
          filename: '[name].[hash][ext]',
          outputPath: (pathData: PathData) => {
            const parts = (pathData.module as Module).context?.split('/')
            return path.join('locales', parts![parts!.length - 1])
          },
        },
      },
      ....
}
```
In my client code I use: 
```ts
const url = `${ASSETS_PATH}/locales/en/${__webpack_hash__}.translations.json`;
```
Does `__webpack_hash__` correspond to what the placeholder `[hash]` resolves in?

**Comment by mborysenko** [comment]
Is there any correlation between placeholders, `[hash]` in `output` of `filename` properties and `__webpack_hash__` variable?

**Comment by mborysenko** [comment]
And DTW is it available in nodejs when SSR is used? For example through `process.env.__webpack_hash__`.

**Reply by alexander-akait** [reply]
`__webpack_hash__` is global variable and you can use it in `SSR`

**Comment by alexander-akait** [comment]
> const url = `${ASSETS_PATH}/locales/en/${__webpack_hash__}.translations.json`;

I don't think it is good idea, why you need it?
